docs/wiki/testing/formal-verification.md

```diff
@@ -0,0 +1,172 @@
+> most abstract form, in all lower pearls one or more concepts used above are explained (refined) in terms of concepts
+> to be explained (refined) in pearls below it, while the bottom pearl eventually explains what still has to be explained
+> in terms of a standard interface (=machine). The family becomes a given collection of pearls that can be strung
+> into a fitting necklace.
+
+Let's write a quick TLA+ spec to model that abstracts a traffic controller.
+
+```bash
+-------------- MODULE TrafficController --------------
+
+CONSTANTS MaxCars
+VARIABLES carsWaiting, greenSignal
+
+Init == /\ carsWaiting = 0
+        /\ greenSignal = FALSE
+
+Arrive(car) == IF carsWaiting < MaxCars THEN carsWaiting' = carsWaiting + 1 ELSE UNCHANGED carsWaiting
+
+Depart == IF carsWaiting > 0 THEN carsWaiting' = carsWaiting - 1 ELSE UNCHANGED carsWaiting
+
+ChangeSignal == /\ carsWaiting > 0
+                /\ greenSignal' = TRUE
+
+Next == \/
+         \E car \in {0, 1}: Arrive(car)
+         \/ Depart
+         \/ ChangeSignal
+
+Invariant == carsWaiting <= MaxCars
+
+Spec == Init /\ [][Next]_<<carsWaiting, greenSignal>> /\ []Invariant
+
+=======================================================
+
+```
+
+Note how this abstraction conveniently ignores all the irrelevant interactions at a traffic signal (honking, anyone?).
+
+**Efficient abstraction is an art.**
+
+Understanding the TLA+ semantics are not important for this discussion. Here is a brief of what it does:
+
+`Init` initializes the system with no cars waiting. The `Arrive` models the arrival of cars, increasing the count of waiting cars if the maximum capacity has not been reached. Conversely, the `Depart` simulates cars departing from the controller, decrementing the count of waiting cars if there are any. Lastly ,`ChangeSignal` dictates that if cars are waiting, the traffic signal switches to green.
+
+The invariant `Invariant == carsWaiting <= MaxCars` ensures the number of cars waiting never exceeds `MaxCars`, a defined constant.
+
+## Ethereum and formal verification
+
+Formal verification plays a critical role in verifying safety and liveliness of Ethereum's decentralized infrastructure and its applications. This rigorous method achieves the highest ranking on the [EAL scale](https://en.wikipedia.org/wiki/Evaluation_Assurance_Level#EAL7:_Formally_Verified_Design_and_Tested), signifying its profound impact on security.
+
+### Protocol verification
+
+[KEVM](https://github.com/runtimeverification/evm-semantics) builds upon [K framework](https://kframework.org/) for crafting formal semantics and conducting verification of the [Ethereum Virtual Machine (EVM)](/wiki/EL/evm.md) specification for correctness.
+
+Formal verification is an essential tool in the test suite and was used to discover a subtle [array-out-of-bound runtime error](https://consensys.io/blog/formal-verification-of-ethereum-2-0-part-1-fixing-the-array-out-of-bound-runtime-error) within the state transition component.
```

Comment:
Runtime verification folks worked on verifying gasper and various parts of the beacon chain. I would list their work here with proper context 
https://github.com/runtimeverification/beacon-chain-verification
https://runtimeverification.com/blog/a-formal-model-in-k-of-the-beacon-chain-ethereum-2-0s-primary-proof-of-stake-blockchain
https://runtimeverification.com/blog/formally-verifying-finality-in-gasper-the-core-of-the-beacon-chain
Also deposit contract, maybe fits the other category https://runtimeverification.com/blog/formal-verification-of-ethereum-2-0-deposit-contract-part-1